### DISCUSSION TITLE
Cannot start on WSL 2

### DISCUSSION THREAD

**Discussion by Tranduy1dol** [zed-industries/zed#14186] [Feature Requests]
Hi, i installed zed in wsl, but i run ```zed``` command nothing happend. How can i fix that?

**Comment by Tranduy1dol** [comment]
when i run ```zed --foreground```, this happend: 
```
thread 'main' panicked at crates/gpui/src/platform/linux/wayland/client.rs:143:51:
called `Result::unwrap()` on an `Err` value: UnsupportedVersion
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Reply by AhmedBaset** [reply]
Same here:

```
thread 'main' panicked at crates/gpui/src/platform/linux/wayland/client.rs:144:51:
called `Result::unwrap()` on an `Err` value: UnsupportedVersion
stack backtrace:
   0:     0x55f6f90ec03b - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h3692694645b1bb6a
   1:     0x55f6f63ce10b - core::fmt::write::h5131d80b4c69b88d
   2:     0x55f6f90e6e63 - std::io::Write::write_fmt::h1fb327a7d8b0eb36
   3:     0x55f6f90ebdae - std::sys_common::backtrace::print::h998d75b840f75a73
   4:     0x55f6f90ed796 - std::panicking::default_hook::{{closure}}::h18ec7fe6a38b9da0
   5:     0x55f6f90ed4b5 - std::panicking::default_hook::hfb3f22c2e4075a6a
   6:     0x55f6f90edc76 - std::panicking::rust_panic_with_hook::h51af00bcb4660c4e
   7:     0x55f6f90edb74 - std::panicking::begin_panic_handler::{{closure}}::h39f76aa863fbe8ce
   8:     0x55f6f90ec509 - std::sys_common::backtrace::__rust_end_short_backtrace::h4d10fc2251b89840
   9:     0x55f6f90ed8d7 - rust_begin_unwind
  10:     0x55f6f5549503 - core::panicking::panic_fmt::h319840fcbcd912ef
  11:     0x55f6f55499f6 - core::result::unwrap_failed::haccb9aaa604e1e21
  12:     0x55f6f75a5b87 - gpui::platform::linux::wayland::client::Globals::new::h7ba30eb8579952c6
  13:     0x55f6f75a9795 - gpui::platform::linux::wayland::client::WaylandClient::new::hbc87361d876b45bb
  14:     0x55f6f75f8c5d - gpui::platform::current_platform::h37cd8dbb34dc3552
  15:     0x55f6f753a145 - gpui::app::App::new::h311ff32ee6b5b11b
  16:     0x55f6fa3deed0 - zed::main::h0f4e0a89c044a0ef
  17:     0x55f6fa3a9223 - std::sys_common::backtrace::__rust_begin_short_backtrace::h2e07cc66a0c11453
  18:     0x55f6fa3a9359 - std::rt::lang_start::{{closure}}::h16085f5a0347278e
  19:     0x55f6f90db843 - std::rt::lang_start_internal::h63a185b0ddd212e9
  20:     0x55f6fa3e335c - main
  21:     0x7f0b38c8cd90 - <unknown>
  22:     0x7f0b38c8ce40 - __libc_start_main
  23:     0x55f6f57f36fe - _start
  24:                0x0 - <unknown>
```

**Reply by Niko-La** [reply]
> when i run `zed --foreground`, this happend:
> 
> ```
> thread 'main' panicked at crates/gpui/src/platform/linux/wayland/client.rs:143:51:
> called `Result::unwrap()` on an `Err` value: UnsupportedVersion
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> ```

Same I am on Windows Arm on Snapdragon X elite.. ughh 

```
/Downloads/zed$ ./zed.app/bin/zed --foreground
thread 'main' panicked at crates/gpui/src/platform/linux/wayland/client.rs:391:49:
called Result::unwrap() on an Err value: NoWaylandLib
note: run with RUST_BACKTRACE=1 environment variable to display a backtrace
```
uname -a
```
nikolawin@BOOK-QEH6FIEMKF:/mnt/c/Users/K/Downloads/zed$ uname -a
Linux BOOK-QEH6FIEMKF 5.15.153.1-microsoft-standard-WSL2 #1 SMP Fri Mar 29 23:16:34 UTC 2024 aarch64 aarch64 aarch64 GNU/Linux
```
neofetch
```
nikolawin@BOOK-QEH6FIEMKF:/mnt/c/Users/Kava/Downloads/zed$ neofetch
            .-/+oossssoo+/-.               nikolawin@BOOK-QEH6FIEMKF 
        :+ssssssssssssssssss+:           ------------------------- 
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu 22.04.4 LTS on Windows 10 aarch64 
    .ossssssssssssssssssdMMMNysssso.       Kernel: 5.15.153.1-microsoft-standard-WSL2 
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Uptime: 32 mins 
  +ssssssssshmydMMMMMMMNddddyssssssss+     Packages: 480 (dpkg), 6 (snap) 
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Shell: bash 5.1.16 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Terminal: Relay(634) 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   CPU: (12) 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   GPU: c454:00:00.0 Microsoft Corporation Device 008a 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   Memory: 506MiB / 7814MiB 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+
.ssssssssdMMMNhsssssssssshNMMMdssssssss.                           
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/                            
  +sssssssssdmydMMMMMMMMddddyssssssss+
   /ssssssssssshdmNNNNmyNMMMMhssssss/
    .ossssssssssssssssssdMMMNysssso.
      -+sssssssssssssssssyyyssss+-
        :+ssssssssssssssssss+:
            .-/+oossssoo+/-.
```

**Comment by dangenendt** [comment]
Same here. Seems like an issue with wayland 🧐
![image](https://github.com/user-attachments/assets/a847e57b-fe80-4971-b0e2-44430c40168d)

**Reply by 4nkitd** [reply]
thread 'main' panicked at crates/gpui/src/platform/linux/wayland/client.rs:143:51:
called `Result::unwrap()` on an `Err` value: UnsupportedVersion
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

**Comment by atomlin** [comment]
Same here. Running on Ubuntu 22:04 in WSL2
```
~$ RUST_BACKTRACE=1 zed --foreground
thread 'main' panicked at crates/gpui/src/platform/linux/wayland/client.rs:143:51:
called `Result::unwrap()` on an `Err` value: UnsupportedVersion
stack backtrace:
   0: rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::result::unwrap_failed
   3: gpui::platform::linux::wayland::client::Globals::new
   4: gpui::platform::linux::wayland::client::WaylandClient::new
   5: gpui::app::App::new
   6: zed::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Reply by AncientPatata** [reply]
```sh
sleep@DESKTOP-M95KURS:~$  RUST_BACKTRACE=1 zed --foreground
thread 'main' panicked at crates/gpui/src/platform/linux/wayland/client.rs:390:49:
called `Result::unwrap()` on an `Err` value: NoWaylandLib
stack backtrace:
   0: rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::result::unwrap_failed
   3: gpui::platform::linux::wayland::client::WaylandClient::new
   4: gpui::app::App::new
   5: zed::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

Same problem, also running Ubuntu 22.04 in WSL2

**Comment by beber89** [comment]
What is distinctive about Zed is that it utilizes the gpu for rendering and it seems that this is the problem here. I am trying to find a way to make it work with my Ryzen AMD graphics on wsl.

**Comment by ayan-cpath** [comment]
Just my 2 cents from a Windows laptop running latest WSL2 + Ubuntu with just an Intel CPU with integrated graphics.

Hope this helps.

```
> RUST_BACKTRACE=full zed --foreground
thread 'main' panicked at crates/gpui/src/platform/linux/wayland/client.rs:143:51:
called `Result::unwrap()` on an `Err` value: UnsupportedVersion
stack backtrace:
   0:     0x55a7c52ca670 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h1e1a1972118942ad
   1:     0x55a7c2c15d2b - core::fmt::write::hc090a2ffd6b28c4a
   2:     0x55a7c52c5323 - std::io::Write::write_fmt::h8898bac6ff039a23
   3:     0x55a7c52ca3ee - std::sys_common::backtrace::print::ha96650907276675e
   4:     0x55a7c52cbd2e - std::panicking::default_hook::{{closure}}::h215c2a0a8346e0e0
   5:     0x55a7c52cba06 - std::panicking::default_hook::h207342be97478370
   6:     0x55a7c52cc206 - std::panicking::rust_panic_with_hook::hac8bdceee1e4fe2c
   7:     0x55a7c52cc104 - std::panicking::begin_panic_handler::{{closure}}::h00d785e82757ce3c
   8:     0x55a7c52cab39 - std::sys_common::backtrace::__rust_end_short_backtrace::h1628d957bcd06996
   9:     0x55a7c52cbe67 - rust_begin_unwind
  10:     0x55a7c1e989b3 - core::panicking::panic_fmt::hdc63834ffaaefae5
  11:     0x55a7c1e98e66 - core::result::unwrap_failed::h82b551e0ff2b2176
  12:     0x55a7c3cd77b7 - gpui::platform::linux::wayland::client::Globals::new::h3bd2fb19603e8a61
  13:     0x55a7c3cdb3f5 - gpui::platform::linux::wayland::client::WaylandClient::new::habf7bfc798f0a0b8
  14:     0x55a7c3c70603 - gpui::app::App::new::h7bc0e2d0ab51826c
  15:     0x55a7c63eca57 - zed::main::h6b345377a7568635
  16:     0x55a7c64ab8f3 - std::sys_common::backtrace::__rust_begin_short_backtrace::h43eabf87c8e5aa31
  17:     0x55a7c660e159 - std::rt::lang_start::{{closure}}::h889d8b7092cc668c
  18:     0x55a7c52ba196 - std::rt::lang_start_internal::h3ed4fe7b2f419135
  19:     0x55a7c63ef26c - main
  20:     0x7f4f55bb5d90 - <unknown>
  21:     0x7f4f55bb5e40 - __libc_start_main
  22:     0x55a7c210594e - _start
  23:                0x0 - <unknown>
  ```

**Reply by 4nkitd** [reply]
thread 'main' panicked at crates/gpui/src/platform/linux/wayland/client.rs:410:49:
called `Result::unwrap()` on an `Err` value: NoCompositor
stack backtrace:
   0:     0x558ddf17303b - <std::sys::backtrace::BacktraceLock::print::DisplayBacktrace as core::fmt::Display>::fmt::h1b9dad2a88e955ff
   1:     0x558ddbd1599b - core::fmt::write::h4b5a1270214bc4a7
   2:     0x558ddf16e1e3 - std::io::Write::write_fmt::hd04af345a50c312d
   3:     0x558ddf1747aa - std::panicking::default_hook::{{closure}}::h96ab15e9936be7ed
   4:     0x558ddf1743f7 - std::panicking::default_hook::h3cacb9c27561ad33
   5:     0x558ddf174e09 - std::panicking::rust_panic_with_hook::hfe205f6954b2c97b
   6:     0x558ddf174d17 - std::panicking::begin_panic_handler::{{closure}}::h6cb44b3a50f28c44
   7:     0x558ddf173509 - std::sys::backtrace::__rust_end_short_backtrace::hf1c1f2a92799bb0e
   8:     0x558ddf1749a4 - rust_begin_unwind
   9:     0x558ddac78853 - core::panicking::panic_fmt::h3d8fc78294164da7
  10:     0x558ddac78d26 - core::result::unwrap_failed::hfa79a499befff387
  11:     0x558ddd08eb2c - gpui::platform::linux::wayland::client::WaylandClient::new::hffb2ff041f6ce5e8
  12:     0x558ddd0e2983 - gpui::platform::current_platform::h5a957bb50e3cb198
  13:     0x558ddd0181a2 - gpui::app::App::new::h6c49728ba6d9597a
  14:     0x558de06f376c - zed::main::h4fe7b4cffd5cb4fb
  15:     0x558de0639953 - std::sys::backtrace::__rust_begin_short_backtrace::hf988928f4655bdef
  16:     0x558de05e8bf9 - std::rt::lang_start::{{closure}}::h2c2bdda00eda9298
  17:     0x558ddf163205 - std::rt::lang_start_internal::h5e7c81cecd7f0954
  18:     0x558de06f8f6c - main
  19:     0x7fae187641ca - <unknown>
  20:     0x7fae1876428b - __libc_start_main
  21:     0x558ddaf4619e - _start
  22:                0x0 - <unknown>

**Comment by bartblokhuis** [comment]
Same here

**Comment by 0x5ea000000** [comment]
it seems that the weston fork version of wslg is too low and not supported, here is a workaround solution to run it in X11 mode instead of wayland
https://github.com/zed-industries/zed/issues/14126#issuecomment-2230014829

**Reply by nowhereman1999** [reply]
marked

**Comment by dragos-cojocaru** [comment]
We need an out-of-the-box experience for Windows + WSL2 + Docker

**Comment by atomlin** [comment]
Is there any update on this?

**Reply by Zander-1024** [reply]
no

**Comment by RyzeNGrind** [comment]
Any updates on this? I would like to migrate away from Cursor and try out Zed on windows. I am currently interested in testing Zed experience inside NixOS-WSL (WSL2). Preferably a clean OOB experience would be nice.

**Reply by iscas-zac** [reply]
I'm also on NixOS. It seems that the native gpu feature is blocked by mesa/dozen as suggested by https://github.com/zed-industries/zed/issues/15295#issuecomment-2583669315. Alternatively, emulated gpu can be used on my computer by `ZED_ALLOW_EMULATED_GPU=1 WAYLAND_DISPLAY='' zeditor`.

**Comment by besendorf** [comment]
Hey,
I tried several of these workaround but could not get zed to run on WSL on Ubuntu 24.04 or Arch. Is anyone still running this on WSL? I have an AMD GPU.

**Comment by ghost** [comment]
Here is how I managed to get it running:

Followed steps described in [this tutorial](https://dev.to/le0_nguyen/how-to-install-zed-editor-on-windows-using-linux-ubuntu-wsl2-step-by-step-guide-3ae0) but _Zed_ didn't start with a GUI.

Next, run: `RUST_BACKTRACE=1 zed --foreground` which mentioned a missing `libasound.so.2`. Installed missing libraries via `sudo apt install libasound2-dev`.

Now _Zed_ starts with a GUI. Packages for Mesa/Vulkan and config options via envvars (as mentioned in the tutorial) seem to be required.

**Comment by med1844** [comment]
I'm trying to run gpui example on [homepage](www.gpui.rs) and saw the same `UnsupportedVersion` error. `WAYLAND_DISPLAY='' cargo run` works in my case.

**Comment by eastLaugh** [comment]
run zed.exe  . instead of zed .

**Reply by creatorrr** [reply]
damn, that worked. thanks!

**Comment by CodeWenjiu** [comment]
Does it represent gpui can not use on wslg?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
